rootfs/api/models/certificate.py

```diff
@@ -179,15 +180,15 @@ def attach_in_kubernetes(self, domain):
             }
 
             secret = self._scheduler._get_secret(app, name).json()['data']
-        except KubeHTTPException:
+        except KubeException:
             self._scheduler._create_secret(app, name, data)
         else:
             # update cert secret to the TLS Ingress format if required
             if secret != data:
                 try:
                     self._scheduler._update_secret(app, name, data)
-                except KubeHTTPException:
-                    raise
+                except KubeException as e:
+                    raise ServiceUnavailable(str(e)) from e
```

Comment:
Yes and no - There is more than one way to skin that cat.

One thing to pay close attention to here is that we are still raising exceptions up to the View, the difference here is we are curating what exceptions are raised, and they contain all the *required* information for the DRF exception handler to translate that into a `HttpResponse` object. In that sense we have not broken the paradigm you described.

When we use the models across one another (app calling build, build calling app, etc) then they are still Python Exceptions that we can catch and process on, however the Models are not Libraries. The scheduler is a library the Models use, and the Models are the Contract between the View and the 3rd party services (scheduler + k8s, logger, docker)

That's how I have been approach it in any case and how I ended up with what I have now